Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/Location/LocationDrawingsTable.tsx

```diff
@@ -0,0 +1,191 @@
+//******************************************************************************************************
+//  LocationDrawings.tsx - Gbtc
+//
+//  Copyright � 2023, Grid Protection Alliance.  All Rights Reserved.
+//
+//  Licensed to the Grid Protection Alliance (GPA) under one or more contributor license agreements. See
+//  the NOTICE file distributed with this work for additional information regarding copyright ownership.
+//  The GPA licenses this file to you under the MIT License (MIT), the "License"; you may not use this
+//  file except in compliance with the License. You may obtain a copy of the License at:
+//
+//      http://opensource.org/licenses/MIT
+//
+//  Unless agreed to in writing, the subject software distributed under the License is distributed on an
+//  "AS-IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. Refer to the
+//  License for the specific language governing permissions and limitations.
+//
+//  Code Modification History:
+//  ----------------------------------------------------------------------------------------------------
+//  11/06/2024 - Collins Self
+//       Generated original version of source code.
+//
+//******************************************************************************************************
+import { SystemCenter } from "@gpa-gemstone/application-typings";
+import { Pencil, TrashCan } from "@gpa-gemstone/gpa-symbols";
+import { GenericController, LoadingScreen, ServerErrorIcon } from "@gpa-gemstone/react-interactive";
+import { ReactTable, Paging } from "@gpa-gemstone/react-table";
+import React from "react";
+import { useAppSelector } from "../hooks";
+import { SelectRoles } from "../Store/UserSettings";
+
+interface IProps {
+    LocationID: number,
+    Edit?: (record: SystemCenter.Types.LocationDrawing) => void,
+    /**
+     * @param UpdateTable Counter that triggers a fetchDrawings function
+     */
+    UpdateTable: number
+}
+
+const LocationDrawingsTable = (props: IProps) => {
+    const [links, setLinks] = React.useState<SystemCenter.Types.LocationDrawing[]>([]);
+    const [sortKey, setSortKey] = React.useState<keyof SystemCenter.Types.LocationDrawing>('Name');
+    const [ascending, setAscending] = React.useState<boolean>(true);
+    const [pageInfo, setPageInfo] = React.useState<{ RecordsPerPage: number, NumberOfPages: number, TotalRecords: number }>({ RecordsPerPage: 0, NumberOfPages: 0, TotalRecords: 0 });
+    const [pageState, setPageState] = React.useState<'error' | 'idle' | 'loading'>('idle');
+    const [page, setPage] = React.useState<number>(0);
+
+    const roles = useAppSelector(SelectRoles); // Deprecated
```

Comment:
why is this here if it is Deprecated?

Reply:
I think I misunderstood what you were saying at one point about moving to controllers over slices, so I thought his was deprecated. The comment has been removed